flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeSchedulerNG.java

```diff
@@ -156,9 +151,7 @@
 
     @Nullable private JobStatusListener jobStatusListener;
 
-    private final RequirementsCalculator requirementsCalculator =
-            new SlotSharingRequirementsCalculator();
-    private final MappingCalculator mappingCalculator = new SlotSharingMappingCalculator();
+    private final SlotSharingSlotAllocator slotAllocator;
```

Comment:
I guess that we need to introduce a `SlotAllocator` interface at some point.

Reply:
there is an interface, i just forgot to use it here